src/Network/Wai/SAML2/Validation.hs

```diff
@@ -51,6 +53,13 @@ validateResponse cfg responseData = runExceptT $ do
     -- get the current time
     now <- liftIO $ getCurrentTime
 
+    (responseXmlDoc, samlResponse) <- ExceptT $ decodeResponse responseData
+    ExceptT $ validateSAMLResponse cfg responseXmlDoc samlResponse now
```

Comment:
It doesn't seem very pretty to have to wrap the results of the calls to `decodeResponse` and `validateSAMLResponse` in `ExceptT` explicitly here, especially since they internally run in `ExceptT` anyway and therefore have to unwrapped first. I suppose the question here is whether an API that's easier to consume within this module or from outside is preferable. 

I'd probably go with having `decodeResponse :: BS.ByteString -> ExceptT SAML2Error IO (XML.Document, Response)` and `validateSAMLResponse :: SAML2Config -> XML.Document -> Response -> UTCTime -> ExceptT SAML2Error IO Assertion` to avoid the unnecessary wrapping/unwrapping in the normal use case (even though there's probably no overhead since `ExceptT` is a `newtype`). What do you think?

Reply:
I stuck to the interface `validateResponse` already has for sake of consistency, but I have no objection to exposing ExceptT either.

Reply:
Thanks for changing this in https://github.com/mbg/wai-saml2/pull/31/commits/afa4bc4a320eb9ac3588710d4e20a9db0abb7135!